tests/contrib/operators/test_opsgenie_alert_operator.py

```diff
@@ -0,0 +1,124 @@
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import unittest
+
+from airflow import DAG, configuration
+
+from airflow.contrib.operators.opsgenie_alert_operator import OpsgenieAlertOperator
+from airflow.utils import timezone
+
+DEFAULT_DATE = timezone.datetime(2017, 1, 1)
+
+
+class TestOpsgenieAlertOperator(unittest.TestCase):
+    _config = {
+        'message': 'An example alert message',
+        'alias': 'Life is too short for no alias',
+        'description': 'Every alert needs a description',
+        'responders': [
+            {'id': '4513b7ea-3b91-438f-b7e4-e3e54af9147c', 'type': 'team'},
+            {'name': 'NOC', 'type': 'team'},
+            {'id': 'bb4d9938-c3c2-455d-aaab-727aa701c0d8', 'type': 'user'},
+            {'username': 'trinity@opsgenie.com', 'type': 'user'},
+            {'id': 'aee8a0de-c80f-4515-a232-501c0bc9d715', 'type': 'escalation'},
+            {'name': 'Nightwatch Escalation', 'type': 'escalation'},
+            {'id': '80564037-1984-4f38-b98e-8a1f662df552', 'type': 'schedule'},
+            {'name': 'First Responders Schedule', 'type': 'schedule'}
+        ],
+        'visibleTo': [
+            {'id': '4513b7ea-3b91-438f-b7e4-e3e54af9147c', 'type': 'team'},
+            {'name': 'rocket_team', 'type': 'team'},
+            {'id': 'bb4d9938-c3c2-455d-aaab-727aa701c0d8', 'type': 'user'},
+            {'username': 'trinity@opsgenie.com', 'type': 'user'}
+        ],
+        'actions': ['Restart', 'AnExampleAction'],
+        'tags': ['OverwriteQuietHours', 'Critical'],
+        'details': {'key1': 'value1', 'key2': 'value2'},
+        'entity': 'An example entity',
+        'source': 'Airflow',
+        'priority': 'P1',
+        'user': 'Jesse',
+        'note': 'Write this down'
+    }
+
+    expected_payload_dict = {
+        'message': _config['message'],
+        'alias': _config['alias'],
+        'description': _config['description'],
+        'responders': _config['responders'],
+        'visibleTo': _config['visibleTo'],
+        'actions': _config['actions'],
+        'tags': _config['tags'],
+        'details': _config['details'],
+        'entity': _config['entity'],
+        'source': _config['source'],
+        'priority': _config['priority'],
+        'user': _config['user'],
+        'note': _config['note']
+    }
+
+    def setUp(self):
+        configuration.load_test_config()
+        args = {
+            'owner': 'airflow',
+            'start_date': DEFAULT_DATE
+        }
+        self.dag = DAG('test_dag_id', default_args=args)
+
+    def test_build_opsgenie_payload(self):
+        # Given / When
+        operator = OpsgenieAlertOperator(
+            task_id='opsgenie_alert_job',
+            dag=self.dag,
+            **self._config
+        )
+
+        payload = operator._build_opsgenie_payload()
+
+        # Then
+        self.assertEqual(self.expected_payload_dict, payload)
+
+    def test_execute(self):
```

Comment:
Since we are making changes

```suggestion
    def test_properties(self):
```

(since this doesn't call the execute() function)

Reply:
Fixed.